### DISCUSSION TITLE
Active battery cooling, custom temperature function?

### DISCUSSION THREAD

**Discussion by w-olsson** [pybamm-team/PyBaMM#2828] [Q&A] [answered]
Hi. 

I would like to include an active cooler into my battery simulations. I belive that the easiest solution in my case would be to implement my own temperature function, is this possible, how do I implement it? 

I am using SPM with electrolyte and have the "thermal: x-lumped" option. 

 If you have other ideas of how to control the battery temperature, please respond to this thread.

**Comment by RuiheLi** [comment, marked as answer]
Hi, to be honest I never tried this before. But one good news is that I just found that the "Ambient temperature [K]" in PyBaMM can actually be a function of time, see [here](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/parameters/thermal_parameters.py#:~:text=def%20T_amb(,%3A%20t%7D)). If you compare that to "Current function [A]" [here](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/parameters/electrical_parameters.py#:~:text=self.current_with_time,.t%7D), you would find that they are very similar. So this folder [PyBaMM/examples/notebooks/parameterization/]([PyBaMM](https://github.com/pybamm-team/PyBaMM)/[examples](https://github.com/pybamm-team/PyBaMM/tree/develop/examples)/[notebooks](https://github.com/pybamm-team/PyBaMM/tree/develop/examples/notebooks)/parameterization/) has lots of useful information on how you can define parameters as a function of time, which may do this job.
